tests/unit/cluster/slot-migration.tcl

```diff
@@ -422,3 +422,24 @@ start_cluster 3 3 {tags {external:skip cluster} overrides {cluster-allow-replica
         resume_process [srv -3 pid]
     }
 }
+
+start_cluster 2 0 {tags {external:skip cluster regression} overrides {cluster-allow-replica-migration no cluster-node-timeout 1000} } {
+    # Issue #563 regression test
+    test "Client blocked on XREADGROUP while stream's slot is migrated" {
+        set stream_name aga
+        set slot 609
+
+        # Start a background process to simulate a blocked client on XREADGROUP
+        R 0 XGROUP CREATE $stream_name mygroup $ MKSTREAM
+        set rd [valkey_deferring_client]
+        $rd xreadgroup GROUP mygroup consumer BLOCK 0 streams $stream_name >
+        after 1000
```

Comment:
```suggestion
```

I don't think this is still needed. I assume it was added previously to give time for the CLI to start, but the deferring client should send the command before the normal client.